lib/lifecycles/bump.js

```diff
@@ -12,8 +12,45 @@ const runLifecycleScript = require('../run-lifecycle-script')
 const semver = require('semver')
 const writeFile = require('../write-file')
 const { resolveUpdaterObjectFromArgument } = require('../updaters')
+const addBangNotes = require('conventional-changelog-conventionalcommits/add-bang-notes')
 let configsToUpdate = {}
 
+function _whatBump(config, commits) {
+  let level = 2
+  let breakings = 0
+  let features = 0
+  let fix = 0
+
+  commits.forEach(commit => {
+    addBangNotes(commit)
+    if (commit.notes.length > 0) {
+      breakings += commit.notes.length
+      level = 0
+    } else if (commit.type === 'feat' || commit.type === 'feature') {
+      features += 1
+      if (level === 2) {
+        level = 1
+      }
+    }
+    if(commit.type === 'fix') {
+      fix += 1
+    }
+  })
+
+  if (config.preMajor && level < 2) {
+    level++
+  }
+
+  if(!breakings && !features && !fix) return {}
+
+  return {
+    level: level,
+    reason: breakings === 1
```

Comment:
I don’t think this check is right- it looks like this is going to be a string?

Reply:
Hmm? I don't get it. You mean`breakings`? `breakings` its a number and will still be a number because length of notes is added. I actually did not write this code just extracted from [here](https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-conventionalcommits/conventionalRecommendedBump.js) that is the `whatBump` function for **conventional-commits**.

Reply:
The only thing that I added it's the fixes tracking to return an empty release if there are neither `breaking`, `feature` or `fix`.